x-pack/plugins/security_solution/server/lib/timeline/routes/utils/compare_timelines_status.ts

```diff
@@ -195,24 +205,27 @@ export class CompareTimelinesStatus {
   }
 
   private get isTemplateVersionConflict() {
-    const version = this.templateTimelineObject?.getVersion;
+    const templateTimelineVersion = this.templateTimelineObject?.getVersion;
     const existingTemplateTimelineVersion = this.templateTimelineObject?.data
       ?.templateTimelineVersion;
     if (
-      version != null &&
+      templateTimelineVersion != null &&
       this.templateTimelineObject.isExists &&
       existingTemplateTimelineVersion != null
     ) {
-      return version <= existingTemplateTimelineVersion;
-    } else if (this.templateTimelineObject.isExists && version == null) {
+      return templateTimelineVersion <= existingTemplateTimelineVersion;
+    } else if (this.templateTimelineObject.isExists && templateTimelineVersion == null) {
       return true;
     }
     return false;
   }
 
   private get isTemplateVersionValid() {
-    const version = this.templateTimelineObject?.getVersion;
-    return typeof version === 'number' && !this.isTemplateVersionConflict;
+    const templateTimelineVersion = this.templateTimelineObject?.getVersion;
+    return (
+      templateTimelineVersion == null ||
+      (isInteger(templateTimelineVersion) && !this.isTemplateVersionConflict)
```

Comment:
If the `templateTimelineVersion` is a string but a number (example: `123`) then the function will return `false`. Is that intended?

Reply:
Yeah, It only takes number as templateTimelineVersion.